contracts/StakeManagerV2.sol

```diff
@@ -306,11 +296,65 @@ contract StakeManagerV2 is
         emit Delegated(preferredOperatorAddress, msg.value);
     }
 
+    /// @dev extracts the withdrawal request and removes from userRequests
+    function _extractRequest(address _user, uint256 _idx) internal returns (WithdrawalRequest storage request) {
+        uint256 numUserRequests = userRequests[_user].length;
+        if (numUserRequests == 0) revert NoWithdrawalRequests();
+        if (_idx >= numUserRequests) revert InvalidIndex();
+
+        uint256 requestId = userRequests[_user][_idx];
+        userRequests[_user][_idx] = userRequests[_user][numUserRequests - 1];
+        userRequests[_user].pop();
+        return withdrawalRequests[requestId];
+    }
+
+    /// @dev internal fn to check if new exchange rate is within limits
+    function _checkIfNewExchangeRateWithinLimits(uint256 currentER) internal {
+        uint256 maxAllowableER = maxExchangeRateSlippageBps * currentER / 10_000;
+        uint256 newER = convertBnbXToBnb(1 ether);
+        if (newER > maxAllowableER) {
+            _pause();
+            revert ExchangeRateTooHigh(currentER, maxAllowableER, newER);
```

Comment:
this will revert so no need to pause here?

Reply:
oh yeah

Reply:
updated the logic, please review